rust/deny.toml

```diff
@@ -0,0 +1,304 @@
+  # Wrapper crates can optionally be specified to allow the crate when it
+  # is a direct dependency of the otherwise banned crate
+  #{ crate = "ansi_term@0.11.0", wrappers = ["this-crate-directly-depends-on-ansi_term"] },
+]
+
+# List of features to allow/deny
+# Each entry the name of a crate and a version range. If version is
+# not specified, all versions will be matched.
+#[[bans.features]]
+#crate = "reqwest"
+# Features to not allow
+#deny = ["json"]
+# Features to allow
+#allow = [
+#    "rustls",
+#    "__rustls",
+#    "__tls",
+#    "hyper-rustls",
+#    "rustls",
+#    "rustls-pemfile",
+#    "rustls-tls-webpki-roots",
+#    "tokio-rustls",
+#    "webpki-roots",
+#]
+# If true, the allowed features must exactly match the enabled feature set. If
+# this is set there is no point setting `deny`
+#exact = true
+
+# Certain crates/versions that will be skipped when doing duplicate detection.
+skip = [
+  "base64",
+  "bitflags",
+  "cocoa",
+  "cocoa-foundation",
+  "core-foundation",
+  "core-graphics",
+  "core-graphics-types",
+  "derive_more",
+  "getrandom",
+  "hashbrown",
+  "heck",
+  "hermit-abi",
+  "indexmap",
+  "itertools",
+  "itoa",
+  "libloading",
+  "nix",
+  "nu-ansi-term",
+  "phf",
+  "phf_codegen",
+  "phf_generator",
+  "phf_macros",
+  "phf_shared",
+  "proc-macro-crate",
+  "quick-xml",
+  "rand",
+  "rand_chacha",
+  "rand_core",
+  "raw-window-handle",
+  "regex-automata",
+  "regex-syntax",
+  "rustls",
+  "syn",
+  "sync_wrapper",
+  "tauri-winrt-notification",
+  "toml",
+  "toml_edit",
+  "tower",
+  "trackable",
+  "wasi",
+  "windows",
+  "windows-core",
+  "windows-implement",
+  "windows-interface",
+  "windows-result",
+  "windows-sys",
+  "windows-targets",
+  "windows_aarch64_gnullvm",
+  "windows_aarch64_msvc",
+  "windows_i686_gnu",
+  "windows_i686_msvc",
+  "windows_x86_64_gnu",
+  "windows_x86_64_gnullvm",
+  "windows_x86_64_msvc",
+  "winnow",
+  "winreg",
+  #"ansi_term@0.11.0",
+  #{ crate = "ansi_term@0.11.0", reason = "you can specify a reason why it can't be updated/removed" },
+]
+# Similarly to `skip` allows you to skip certain crates during duplicate
+# detection. Unlike skip, it also includes the entire tree of transitive
+# dependencies starting at the specified crate, up to a certain depth, which is
+# by default infinite.
+skip-tree = [
+  #"ansi_term@0.11.0", # will be skipped along with _all_ of its direct and transitive dependencies
+  #{ crate = "ansi_term@0.11.0", depth = 20 },
+]
+
+# This section is considered when running `cargo deny check sources`.
+# More documentation about the 'sources' section can be found here:
+# https://embarkstudios.github.io/cargo-deny/checks/sources/cfg.html
+[sources]
+# Lint level for what to happen when a crate from a crate registry that is not
+# in the allow list is encountered
+unknown-registry = "warn"
+# Lint level for what to happen when a crate from a git repository that is not
+# in the allow list is encountered
+unknown-git = "allow"
```

Comment:
This is also different from the default config, normally `git` dependencies are banned but we make quite liberal use of them so I opted to set it to `allow`.